GitCommands/Git/GetAllChangedFilesOutputParser.cs

```diff
@@ -92,19 +92,21 @@ internal List<GitItemStatus> GetAllChangedFilesFromString_v1(string getAllChange
                 }
 
                 char x = status[0];
-                char y = status.Length > 1 ? status[1] : ' ';
+                char y = status.Length > 1 ? status[1] : GitItemStatusConverter.UnmodifiedStatus_v1;
 
-                if (fromDiff && staged == StagedStatus.WorkTree && x == 'U')
+                if (fromDiff && staged == StagedStatus.WorkTree && x == GitItemStatusConverter.UnmergedStatus)
                 {
                     // git-diff has two lines to inform that a file is modified and has a merge conflict
                     continue;
                 }
 
-                if (x != '?' && x != '!' && x != ' ')
+                // Skip unmerged where both are modified: Only worktree interesting.
+                if ((x != GitItemStatusConverter.UntrackedStatus && x != GitItemStatusConverter.IgnoredStatus && x != GitItemStatusConverter.UnmodifiedStatus_v1)
+                    || !(x == GitItemStatusConverter.UnmergedStatus && y == GitItemStatusConverter.UnmergedStatus))
```

Comment:
more similar to the change at line 248
```suggestion
                    || x != GitItemStatusConverter.UnmergedStatus || y != GitItemStatusConverter.UnmergedStatus)
```